### PR TITLE
merge: add pkg-config file

### PR DESCRIPTION
**Pull request by raphi011**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by raphi011** [comment]
This does not work yet because if I `src/include/CMakeLists.txt` explicitly says `kdbconfig.h` should not be installed but `kdblogger.h` depends on it and fails the build:

Any advice @markus2330 ?

```sh
env PKG_CONFIG_PATH=/usr/local/lib/pkgconfig go build
# go.libelektra.org/merge
In file included from /usr/local/include/elektra/kdberrors.h:14,
                 from /usr/local/include/elektra/kdbmerge.h:13,
                 from ./merge.go:4:
/usr/local/include/elektra/kdblogger.h:12:10: fatal error: kdbconfig.h: No such file or directory
   12 | #include "kdbconfig.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
```

File: src/include/CMakeLists.txt [hunk 1]
```diff
@@ -186,6 +186,9 @@ install (FILES kdbextension.h
 	       elektra.h
 	       kdbendian.h
 	       kdbmacros.h
+		   kdbmerge.h
+		   kdberrors.h
```
Comment (markus2330) [inline src/include/CMakeLists.txt]: This is a private file and should not be installed. @Chemin1 can you fix that kdbmerge.h does not include kdberrors.h nor kdblogger.h.

Comment (Chemin1) [reply src/include/CMakeLists.txt]: Fixed in #3234

**Comment by Chemin1** [comment]
@raphi011 I think Jenkins will fail unless you add release notes to the PR.

**Comment by raphi011** [comment]
> @raphi011 I think Jenkins will fail unless you add release notes to the PR.

I know, I will add one as soon as your PR that removes the private header files are merged.

**Comment by raphi011** [comment]
jenkins build libelektra please

**Comment by raphi011** [comment]
Weird [build failure](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/PR-3241/2/pipeline/)

`Caught the following exception: null`

**Comment by raphi011** [comment]
jenkins build libelektra please

**Comment by raphi011** [comment]
jenkins build libelektra please

**Comment by raphi011** [comment]
ready to merge

**Comment by raphi011** [comment]
fixed the conflict

**Comment by markus2330** [comment]
Lets hope the build server passes...

**Comment by raphi011** [comment]
@markus2330 all green

**Comment by markus2330** [comment]
@raphi011 thank you very much! Please also set "ready to merge" if ready to merge (i.e. you do not plan any changes and the build server passed).

**Pull request merged**